valhalla/midgard/encoded.h

```diff
@@ -88,7 +88,7 @@ std::string encode(const container_t& points, const int precision = 1e6) {
   // handy lambda to turn an integer into an encoded string
   auto serialize = [&output](int number) {
     // move the bits left 1 position and flip all the bits if it was a negative number
-    number = number < 0 ? ~(number << 1) : (number << 1);
+    number = number < 0 ? ~(static_cast<unsigned int>(number) << 1) : (number << 1);
```

Comment:
this is again bit shifting a signed number and in this case we know its negative. im surprised this doesnt change lots of stuff.. one good test for this would be to run valhalla_export_edges on a larger tileset since it dumps out the whole tilesets shape. then we can look at the diffs in there to see what this does. maybe we were just lucky on the platforms that we run that the implementation dependent handling was equivalent to the above? we need to test this more rigorously

Reply:
Even more lucky, because `implementation dependent` is for right shift of negative value, for left it's UB.
What exactly do you suggest to test? As for correctness, there is a unit-test eg [this one](https://github.com/valhalla/valhalla/blob/master/test/encode.cc#L393) which does encoding-decoding and definitely creates the case with negative values.

Reply:
well my suggestion was as i stated:

> one good test for this would be to run valhalla_export_edges on a larger tileset ...

yep i am aware of the unit test (i wrote the original), but its not extensive. the reason i want to do a broader test is because this touches literally everything. every single route, mapmatch, how the data is stored. this is a fundamental change so we better be sure its ok. same with the other one above

Reply:
i wrote a small program just to check `int << 1` vs `static_cast<unisigned int>(int) << 1` and it seems that at least on my computer the two operations have identical results.

Reply:
@kevinkreiser I did the testing we discussed
1. Build tiles
2. export edges -> compare
3. Add live traffic -> compare tiles
for Utrecht tiles. Comparison showed no difference. Is there any testing to be done, eg regarding [this comment](https://github.com/valhalla/valhalla/pull/2498#discussion_r463638921) ?